src/FloatingActionButton/FloatingActionButton.js

```diff
@@ -166,6 +166,7 @@ class FloatingActionButton extends Component {
   componentWillReceiveProps(nextProps) {
     if (nextProps.disabled !== this.props.disabled) {
       this.setState({
+        hovered: nextProps.disabled ? false : this.state.hovered,
```

Comment:
Using `the.state` seems dangerous based on this comment https://github.com/callemall/material-ui/issues/2983#issuecomment-231541664.